tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/EnforceRateLimitingPlan.scala

```diff
@@ -0,0 +1,70 @@
+package com.linagora.tmail.mailets
+
+import com.linagora.tmail.rate.limiter.api.{OperationLimitations, RateLimitationPlanRepository, RateLimitingPlan, RateLimitingPlanUserRepository}
+import org.apache.james.core.Username
+import org.apache.james.rate.limiter.api.{AcceptableRate, RateExceeded, RateLimiterFactory, RateLimitingResult, Rule, Rules}
+import org.apache.mailet.Mail
+import org.apache.mailet.base.GenericMailet
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import javax.inject.Inject
+
+class EnforceRateLimitingPlan @Inject()(planRateLimiterResolver: PlanRateLimiterResolver) extends GenericMailet {
+
+  private var operationLimitation: String = _
+  private var exceededProcessor: String = _
+
+  override def init(): Unit = {
+    operationLimitation = getInitParameter("operationLimitation")
+    exceededProcessor = getInitParameter("exceededProcessor", Mail.ERROR)
+  }
+
+  override def service(mail: Mail): Unit =
+    mail.getMaybeSender.asOptional()
+      .ifPresent(sender => {
```

Comment:
Recipients should be used to enforce delivery limits, if they are local

Reply:
Should we have more parameters to declare it? 

Example:
```
  var rateLimitSender: Boolean = getInitParameter("rateLimitSender", false)
  var rateLimitRecipient: Boolean = getInitParameter("rateLimitRecipient", false)
```

IMO it will more flexible than implicit by name of "operationLimitation" ([transit,relay,local-delivery])